addons/purchase_stock/models/product.py

```diff
@@ -49,7 +84,6 @@ def _compute_monthly_demand(self):
         move_qty_by_products = self.env['stock.move']._read_group(move_domain, ['product_id'], ['product_qty:sum'])
         qty_by_product = {product.id: qty for product, qty in move_qty_by_products}
 
-        based_on = self.env.context.get("suggest_based_on", "this_month")
         factor = 1
         if based_on == "one_year":
```

Comment:
It would be great to have everything base on date. I would compute the number of days base on the relative delta between both date and compute the factor base on that